packages/grafana-prometheus/src/configuration/AlertingSettingsOverhaul.tsx

```diff
@@ -43,7 +44,7 @@ export function AlertingSettingsOverhaul<T extends AlertingConfig>({
               className={styles.switchField}
             >
               <Switch
-                value={options.jsonData.manageAlerts !== false}
+                value={options.jsonData.manageAlerts ?? config.defaultDatasourceManageAlertsUiToggle}
```

Comment:
For context: when `options.jsonData.manageAlerts` is nullish (and that happens when it is not yet configured), we use the `config.defaultDatasourceManageAlertsUiToggle` setting.